kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -13431,6 +13454,13 @@ public void setOldDisplayName(String oldDisplayName) {
 		public void open();
 
 		/**
+		 * Get a direct link to the asset so it doesn't have to be streamed.
+		 * @param resource
+		 * @return URI or null if no direct link is available
+		 */
+		public URI getDirectLink(ContentResource resource);
```

Comment:
I like the Optional<URI> suggestion a lot .... I'll do it as a sub-JIRA once I'm satisfied this code works well on Java 7 .....